recipe/activate.sh

```diff
@@ -22,18 +22,34 @@ if [ ! -w "$WRAPPER_DIR" ]; then
     return
 fi
 
-# there's only one lib in that folder, but the libname changes
+# Everything that follows amounts to very carefully creating the
+# symlink if need be.
+
+# there's only one lib in the placeholder folder, but the libname changes
 # based on the version so use a loop instead of hardcoding it.
-for f in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
-    if [ ! -e "$f" ]; then
+for target in "$GDB_PREFIX/$PLACEHOLDER/lib/"*.py; do
+    if [ ! -e "$target" ]; then
         # If the file doesn't exist, skip this iteration of the loop.
         # (This happens when no files are found, in which case the
         # loop runs with f equal to the pattern itself.)
         continue
     fi
-    target="$WRAPPER_DIR/$(basename "$f")"
-    # We have write permissions to WRAPPER_DIR but not necessarily target.
-    # Thus it's safest to delete the target in case it already exists.
-    rm -f "$target"
-    ln -s "$f" "$WRAPPER_DIR/$(basename "$f")"
+    symlink="$WRAPPER_DIR/$(basename "$target")"
+    # Check if symbolic link already exists and points to correct file
+    if [ -L "$symlink" ] && [ "$(readlink "$symlink")" = "$target" ]; then
```

Comment:
This is the most important part. Covers every activation after the link is created